Source/EasyNetQ/Producer/PersistentChannel.cs

```diff
@@ -115,25 +160,56 @@ private void OnReturn(object sender, BasicReturnEventArgs args)
 
         private void OnAck(object sender, BasicAckEventArgs args)
         {
-            eventBus.Publish(MessageConfirmationEvent.Ack((IModel)sender, args.DeliveryTag, args.Multiple));
+            eventBus.Publish(MessageConfirmationEvent.Ack((IModel) sender, args.DeliveryTag, args.Multiple));
         }
 
         private void OnNack(object sender, BasicNackEventArgs args)
         {
-            eventBus.Publish(MessageConfirmationEvent.Nack((IModel)sender, args.DeliveryTag, args.Multiple));
+            eventBus.Publish(MessageConfirmationEvent.Nack((IModel) sender, args.DeliveryTag, args.Multiple));
         }
 
-        private static bool NeedRethrow(OperationInterruptedException exception)
+        private static ExceptionVerdict GetExceptionVerdict(Exception exception)
         {
-            try
+            switch (exception)
             {
-                var amqpException = AmqpExceptionGrammar.ParseExceptionString(exception.Message);
-                return amqpException.Code != AmqpException.ConnectionClosed;
+                case OperationInterruptedException e:
+                    return e.ShutdownReason?.ReplyCode switch
+                    {
+                        AmqpErrorCodes.ConnectionClosed => ExceptionVerdict.Suppress,
+                        AmqpErrorCodes.AccessRefused => ExceptionVerdict.ThrowWithChannelClosure,
+                        AmqpErrorCodes.NotFound => ExceptionVerdict.ThrowWithChannelClosure,
+                        AmqpErrorCodes.ResourceLocked => ExceptionVerdict.ThrowWithChannelClosure,
+                        AmqpErrorCodes.PreconditionFailed => ExceptionVerdict.ThrowWithChannelClosure,
+                        _ => ExceptionVerdict.Throw
+                    };
+                case NotSupportedException e:
+                    var isRequestPipeliningForbiddenException = e.Message.Contains(RequestPipeliningForbiddenMessage);
+                    return isRequestPipeliningForbiddenException
+                        ? ExceptionVerdict.SuppressWithChannelClosure
+                        : ExceptionVerdict.Throw;
+                case EasyNetQException _:
+                    return ExceptionVerdict.Suppress;
+                default:
+                    return ExceptionVerdict.Throw;
             }
-            catch (ParseException)
+        }
+
+        private readonly struct ExceptionVerdict
+        {
+            private ExceptionVerdict(bool rethrow, bool closeChannel)
             {
-                return true;
+                Rethrow = rethrow;
+                CloseChannel = closeChannel;
             }
+
+            public static ExceptionVerdict Suppress { get; } = new ExceptionVerdict(false, false);
+            public static ExceptionVerdict Throw { get; } = new ExceptionVerdict(true, false);
+            public static ExceptionVerdict SuppressWithChannelClosure { get; } = new ExceptionVerdict(false, true);
```

Comment:
`SuppressWithChannelClosure` -> `SuppressAndCloseChannel` ? like properties named as verbs: `Suppress` and `Throw `

Reply:
Fixed. Thanks.